pydicom/tests/test_uid.py

```diff
@@ -187,10 +184,15 @@ def test_name(self):
     def test_name_with_equal_hash(self):
         """Test that UID name works for UID with same hash as predefined UID.
         """
```

Comment:
I am aware that I'm not fluent enough with pydicom codebase. But it took me long time to understand this test and I'm still not sure that I do. Several things that would had helped me to understand it would had writing with the following changes:
```diff
- original_hash = UID.__hash__
- UID.__hash__ = lambda val: hash(JPEGLSLossy)
+ original_uid_hash_funtion = UID.__hash__
+ UID.__hash__ = lambda _: 88888888888
```
In this manner, using `original_uid_hash_funtion` makes splicit that I'm storing a function to restore afterwards and not a value. And using,  `lambda _: 88888888888` as the new `__hash__` function is clear that I'm using a constant arbitrary value.

Nevertheless I think that this way of monkeypatching should be avoided all together. Two solutions: 

#### Using Mock
The problem of using mock is that its not part of unittest in python 2.7. Therefore either we add mock as a dependence or we vendor it. 

```py
from mock import patch
with patch('pydicom.uid.UID') as MockClass:
    instance = MockClass('1.2.3')
    instance.__hash__.return_value = 42
    assert instance == '1.2.3'
```
#### Temporal subclassing
Create a class in the scope of the testing function that modifies the `__hash__` behavior
```py
class mockedUID(UID):
    def __hash__(self):
        return 42

uid = mockedUID('1.2.3')
assert uid.name == '1.2.3'
```

Reply:
> And using, lambda _: 88888888888 as the new `__hash__` function is clear that I'm using a constant arbitrary value.

No, we need to use the hash of an existing uid.
> Using Mock

I would like to avoid adding an additional dependency as long as it is used in only one place.
I will go with the second approach, it is indeed better readable.